libs/features/placeholders.js

```diff
@@ -1,30 +1,94 @@
-const placeholders = {};
-let fetched = false;
+const cachedPlaceholders = {};
 
 // eslint-disable-next-line no-async-promise-executor
-const fetchPlaceholders = (config) => new Promise(async (resolve) => {
-  if (fetched) { resolve(placeholders); return; }
+const fetchPlaceholders = (config, sheet) => new Promise(async (resolve) => {
+  if (cachedPlaceholders?.[config.locale.ietf]?.[sheet]) {
+    resolve(cachedPlaceholders[config.locale.ietf][sheet]);
+    return;
+  }
+
   const path = `${config.locale.contentRoot}/placeholders.json`;
-  const resp = await fetch(path);
+  const query = sheet !== 'default' && typeof sheet === 'string' && sheet.length ? `?sheet=${sheet}` : '';
+  const resp = await fetch(`${path}${query}`);
   const json = resp.ok ? await resp.json() : { data: [] };
-  if (json.data.length === 0) { resolve(placeholders); return; }
+  if (json.data.length === 0) { resolve({}); return; }
+  cachedPlaceholders[config.locale.ietf] = cachedPlaceholders[config.locale.ietf] || {};
+  cachedPlaceholders[config.locale.ietf][sheet] = {};
   json.data.forEach((item) => {
-    placeholders[item.key] = item.value;
+    cachedPlaceholders[config.locale.ietf][sheet][item.key] = item.value;
   });
-  fetched = true;
-  resolve(placeholders);
+  resolve(cachedPlaceholders[config.locale.ietf][sheet]);
 });
 
-function findPlaceholder(suppliedPlaceholders, key) {
-  return suppliedPlaceholders[key] || key.replaceAll('-', ' ');
+function keyToStr(key) {
+  return key.replaceAll('-', ' ');
 }
 
-export async function replaceText(config, regex, text) {
-  const fetchedPlaceholders = await fetchPlaceholders(config);
-  return text.replaceAll(regex, (_, key) => findPlaceholder(fetchedPlaceholders, key));
+async function getPlaceholder(key, config, sheet) {
+  const defaultLocale = 'en-US';
+
+  const getDefaultPlaceholders = async () => {
+    // Maybe we should modify the getConfig() method to provide a default config?
+    // Could contentRoot be different than the origin?
+    const defaultConfig = {
+      locale: {
+        ietf: defaultLocale,
+        contentRoot: window.location.origin,
```

Comment:
I'm not really sure this is the correct approach, I'm not aware of the various use-cases for `contentRoot`.

Reply:
Hmm, this is a tricky one.

One scenario where this would fail is if we used placeholders for the Profile which can be used standalone, I would guess this wouldn't work anymore then.

I'd say it's fine to solve that once there's a use-case though

Reply:
@overmyheadandbody you're spot on that the origin can be different from the contentRoot. In [stock](https://github.com/adobecom/stock/blob/3c631b44403019087606321c4cd4a28e428e9834/pages/scripts/scripts.js#L28) for example, the contentRoot is under pages